### DISCUSSION TITLE
Does anyone here have an example about loading when we have a pagination?

### DISCUSSION THREAD

**Discussion by singleseeker** [vercel/next.js#51342] [Help]
### Summary

With loading.js, it's easy to show the effect when you init your page, but what about the pagination?

Imagine that you have pagination, and you change the page by 

```
onClick = {
() => router.push(`${pathname}?page=xx`);
}
```
What should we do?

**Comment by SamKomesarook** [comment]
Consider wrapping the children on the page that require fetched data in a [suspense boundary](https://react.dev/reference/react/Suspense). It will display the fallback until the children have loaded.

**Reply by singleseeker** [reply]
Yep, I've tried, but I found that it cost a long time to see the loading state after you click the page button.
It's a litter weird; I cannot feature it.
